.github/workflows/main.yml

```
@@ -0,0 +1,43 @@
+#main.yml
+
+name: Main
+on: pull_request
+
+jobs:
+  checks:
+    runs-on: macOS-latest
+    steps:
+
+      - name: Project Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Compute key for CI cache
+        run: ./.github/workflows/compute-ci-cache-key.sh key.txt
+
+      - name: Setup cache
+        uses: actions/cache@v2.1.3
+        continue-on-error: true
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('key.txt') }}
+          restore-keys: ${{ runner.os }}-gradle-
+
+      - name: Setup JDK 1.8
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: 1.8
+
+      - name: Clean project
+        run: ./gradlew clean
+
+      - name: Build project
+        run: ./gradlew build --stacktrace
+
+      - name: Assemble Test APK
+        run: ./gradlew assembleAndroidTest --stacktrace
+
+      - name: Espresso Checks
+        uses: reactivecircus/android-emulator-runner@v2.13.0
+        with:
+          api-level: 23
+          script: ./espresso-checks.sh
```

I would rather inline these, since it makes this script more readable (don't have to go open another file), and it's just two lines anyway. You can use the `|` character to specify multiple lines, although both tasks could be specified in the same command.
```suggestion
          script: ./gradlew leakcanary-android-core:connectedCheck leakcanary-android-instrumentation:connectedCheck --stacktrace
```

---

Inlined here https://github.com/square/leakcanary/pull/2012/commits/e189e68aca3f4276a771c2072f26be8882a527e4